### DISCUSSION TITLE
How long does backpressure "stick around" on a sink?

### DISCUSSION THREAD

**Discussion by davidpellcb** [vectordotdev/vector#21652] [Q&A] [answered]
We're running Agents and Aggregators in Kubernetes (http sink, http_server source). In most clusters the majority of log volume comes from a few nodes/Agents. Our Aggregators scale out with a Horizontal Pod Autoscaler as average CPU goes up but in general a handful of our Aggregators are getting blasted while the rest aren't doing all that much.

If an one of these hot Agents receives backpressure because it's blasting a specific Aggregator, when max_connection_age_secs is reached on the http_server source and the Agent disconnects to find a new target, will that backpressure from the previous Aggregator hang around? Or will it immediately get feedback from the next Aggregator it sends to (assuming it's one that wasn't just getting blasted by another Agent) that it can speed up sending again?

**Comment by jszwedko** [comment, marked as answer]
The backoff state will be persisted to the new aggregator, but should recover (hopefully quickly) as it receives faster responses.
